src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -221,6 +239,13 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                         routeValues,
                         method: null,
                         htmlAttributes: null);
+
+                    //Append the fragment to action
+                    if (Fragment != null)
```

Comment:
Given that you'll probably end up doing this for the HtmlHelper it's probably worth designing a way to do this that works for both HtmlHelpers and TagHelpers (extension method that accepts a fragment)